File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    answer = 0
    dp = []
    for i in range(len(triangle)):
        dp.append([0]*(i+1))
    for i in range(len(triangle[-1])):
        dp[-1][i] = triangle[-1][i]
    for y in range(len(triangle)-2,-1,-1):
        for x in range(len(triangle[y])):
            dp[y][x] = max(triangle[y][x]+dp[y+1][x], triangle[y][x]+dp[y+1][x+1])
    return dp[0][0]

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.js
function solution(triangle) {
  for (let y = triangle.length - 2; y >= 0; y--) {
    const row = triangle[y];
    const next = triangle[y + 1];
    for (let x = 0, len = row.length; x < len; x++) {
      const a = next[x];
      const b = next[x + 1];
      row[x] += (a > b ? a : b);
    }
  }
  return triangle[0][0];
}